keep/api/routes/alerts.py

```diff
@@ -257,43 +262,52 @@ def assign_alert(
     return {"status": "ok"}
 
 
-def discard_task(
+def discard_future(
     trace_id: str,
-    task: asyncio.Task,
+    future: Future,
     running_tasks: set,
     started_time: float,
 ):
     try:
-        running_tasks.discard(task)
-        running_tasks_gauge.dec()  # Decrease total counter
-        running_tasks_by_process_gauge.labels(
-            pid=os.getpid()
-        ).dec()  # Decrease process counter
-
-        # Log any exception that occurred in the task
-        if task.exception():
-            logger.error(
-                "Task failed with exception",
-                extra={
-                    "trace_id": trace_id,
-                    "error": str(task.exception()),
-                    "processing_time": time.time() - started_time,
-                },
-            )
-        else:
+        running_tasks.discard(future)
+        running_tasks_gauge.dec()
+        running_tasks_by_process_gauge.labels(pid=os.getpid()).dec()
+
+        # Log any exception that occurred in the future
+        try:
+            exception = future.exception()
+            if exception:
+                logger.error(
+                    "Task failed with exception",
+                    extra={
+                        "trace_id": trace_id,
+                        "error": str(exception),
+                        "processing_time": time.time() - started_time,
+                    },
+                )
+            else:
+                logger.info(
+                    "Task completed",
+                    extra={
+                        "processing_time": time.time() - started_time,
+                        "trace_id": trace_id,
+                    },
+                )
+        except concurrent.futures.CancelledError:
             logger.info(
```

Comment:
this should never happen i guess, right? lets add an alert
```suggestion
            logger.error(
```